src/Adapter/Container/ConfigYamlLoader.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Container;
+
+use Symfony\Component\Config\Loader\FileLoader;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * ConfigYamlLoader is used to load YAML config files and get the config as array.
+ */
+class ConfigYamlLoader extends FileLoader
```

Comment:
This class looks pretty generic. Doesn't symfony already handle this stuff?

Reply:
Not for config files only, they have `Symfony\Component\DependencyInjection\Loader\YamlFileLoader` but it's meant to loader services and needs a container as input
The `parseImports` from this class is inspired by their `YamlFileLoader` but I needed something more simple that simply loads a config YAML files along with its imported files, and I couldn't find any class that did this (based on a search on the abstract `FileLoader` class)

Reply:
Maybe we could move this in `PrestaShopBundle\DependencyInjection\Config` ?